src/v_video.c

```diff
@@ -849,9 +871,39 @@ void V_ScreenShot(char *format)
 #ifdef HAVE_LIBPNG
     if (png_screenshots)
     {
-    WritePNGfile(lbmname, I_VideoBuffer,
-                 SCREENWIDTH, SCREENHEIGHT,
-                 W_CacheLumpName (DEH_String("PLAYPAL"), PU_CACHE));
+        byte * buf, * oldbuf;
+        int oldpitch;
+
+        // Prepare a temporary buffer to write the screenshot to
+        buf = malloc(screenshot_width * screenshot_height);
+
+        if(!buf)
+        {
+            return;
+        }
+
+        // Ideally we'd do this only once, but V_Init is called too early
+        // and screenshot_width/height haven't been read from the config
+        V_InitScreenShotScaler();
+
+        // back up the scaling system's current output buffer settings
+        oldbuf = I_GetScaleDestBuf();
```

Comment:
Hi, sorry for long delay in getting back to you. I cleaned up the style issues you pointed out before I went on a holiday and forgot to push it when I got back.

As for the wrapper approach: hmm. Well, I can see that would be attractive if you wanted to minimise the changes made to the scaler code. And I recognise you would want to minimise the changes made to vanilla doom code. But afaik the scaler code is all new to chocolate, so I don't see why you would want to minimise changes to it. I'd be concerned that adding a wrapper is causing complexity by accretion. In general, moving the code away from globals is attractive to me, but not for any objective reason that matters to chocolate-doom (re-entrant, parallel execution and reasoning etc. - not relevant here).

At the end of the day, it's your project, so I will try to do what I can to do something that satisfied you. Take my thoughts with as much salt as you like :)